structure_tests/structure_test_v1.go

```diff
@@ -107,3 +120,45 @@ func (st StructureTestv1) RunFileContentTests(t *testing.T) {
 		}
 	}
 }
+
+// Given a list of environment variables and a list of lists of strings,
+// retrieve each environment variable's value and replace all occurrences
+// of it in each list of strings provided.
+func SubstituteEnvVars(t *testing.T, vars []string, lists ...*[]string) {
+	for _, env_var := range vars {
+		value := os.Getenv(env_var)
+		if value == "" {
+			t.Fatalf("Variable %s not found in environment!", env_var)
+		}
+		for _, list := range lists {
+			for i := range *list {
+				str := (*list)[i]
+				(*list)[i] = strings.Replace(str, "$"+env_var, value, -1)
+			}
+		}
+	}
+}
+
+func ProcessCommands(t *testing.T, commands []string) {
```

Comment:
What about changing both so it's like Kubernetes, basically the way you have setup:

setup: ["foo", "bar"]

then the test command could be the same, instead of splitting up command and flags.